### PR TITLE
8273514: java/util/DoubleStreamSums/CompensatedSums.java failure

### PR DESCRIPTION
**Pull request by igraves**
Relaxing some assertion bounds to allow for small error values that still show improvement over previous summation method.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8273514](https://bugs.openjdk.java.net/browse/JDK-8273514): java/util/DoubleStreamSums/CompensatedSums.java failure

### Reviewers
 * [Roger Riggs](https://openjdk.java.net/census#rriggs) (@RogerRiggs - **Reviewer**) ⚠️ Review applies to 8f77f3d374df8d36278c613269a1439c2392b807
 * [Joe Darcy](https://openjdk.java.net/census#darcy) (@jddarcy - **Reviewer**) ⚠️ Review applies to 8f77f3d374df8d36278c613269a1439c2392b807

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/5430/head:pull/5430` \
`$ git checkout pull/5430`

Update a local copy of the PR: \
`$ git checkout pull/5430` \
`$ git pull https://git.openjdk.java.net/jdk pull/5430/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 5430`

View PR using the GUI difftool: \
`$ git pr show -t 5430`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/5430.diff">https://git.openjdk.java.net/jdk/pull/5430.diff</a>

</details>

### CODE REVIEW

**Review by RogerRiggs — APPROVED** [review]
LGTM

File: test/jdk/java/util/DoubleStreamSums/CompensatedSums.java [hunk 1]
```diff
@@ -87,12 +87,10 @@ public void testCompensatedSums() {
```
Comment (jddarcy) [inline test/jdk/java/util/DoubleStreamSums/CompensatedSums.java:87]: Above, if there are going to be multiple uese of Math.pow(arg, 2) in the test, I recommend defining a local private static square(double d) method for this test where square is 
double square(double arg) {return arg * arg;}
The library method pow(arg, 2) does a check for the exponent being two and does a multiple instead.

**Review by jddarcy — APPROVED** [review]

File: test/jdk/java/util/DoubleStreamSums/CompensatedSums.java (hunk 1 above)
Comment (igraves) [reply test/jdk/java/util/DoubleStreamSums/CompensatedSums.java:84]: Good call. Done.

**Pull request closed without merging**